contents/data/databases.md

```diff
@@ -0,0 +1,115 @@
+# Introduction to Databases & Database Management Systems (DBMS)
+
+Authors: [Syed Abdullah](https://github.com/Skaty)
+
+## Why learn databases and DBMSes?
+
+The programs that we create would handle data in some way or another. Be it a simple calculator application that tabulates user calculations (and maybe store them in a log of recent calculation) or a cluster of servers that handle indexing of a large number of documents. Data is what is worked on by any program, the inputs and outputs of a program or even a simple function are data.
+
+As our programs get larger and the data that is worked on becomes much more complex, there needs to be a way for us to systematically store and access data that is being worked on. It could be a crude structure, designed specifically for a particular use case or something that is more standardised, for instance, an application that handles data.
+
+This notion of structuring data and providing an easy to use abstraction for accessing, storing and performing mundane operations on data (for instance, sorting them) is what this guide will cover.
+
+## Introduction
+
+![DB-DBMS Relationship](images/db-dbms-relation.png)
+
+A **database** is a collection of data. While any data in a medium can theoretically be a database (for instance, scribbling on a piece of paper), databases in the context of this guide typically involve data that are organised in some manner.
+
+Programs generally do not access the raw data directly. Instead, a  **database management system (DBMS)** is used, which will handle the storage, retrieval and updating of the data.
+
+Sometimes people use the term database to refer to a DBMS. However, to avoid any confusion, this guide will use these terms as defined above.
+
+There are various concepts in the field of databases and DBMS and this guide will cover the basic concepts that are useful for someone who is starting out.
+
+## Database models
+
+For databases to make any sense, there has to be a certain logical structure in the database (for instance, how data is stored). This logical structure is known as a **database model**.
+
+There are many types of database models. Some examples are:
+
+- [**Relational model**](https://en.wikipedia.org/wiki/Relational_model) - the most widely used database model, usually modeled using a table format. The data structure is defined by a set of relations and domains (types) that dictates the constraints. Constraints are also established for the type of operations that can be done.
+- [**Document-oriented**](https://en.wikipedia.org/wiki/Document-oriented_database) - data is stored in documents, that encapsulates the data. This data is normally stored in a semi-structured manner.
```

Comment:
seems like this article is covering NoSQL systems. There are other other categories such as Extensible Records, Key-Value and Graph data stores, which can also be placed here.